packages/gatsby/src/redux/actions.js

```diff
@@ -1099,13 +1099,18 @@ actions.createRedirect = ({
     pathPrefix = store.getState().config.pathPrefix
   }
 
+  // Check if URL is an external link
+  // and if it is, don't add pathPrefix
+  const isExternalUrl = toPath.startsWith(`https://` || `http://`)
```

Comment:
Is it possible for redirect paths to begin with `//`, and if so is this handled correctly?

Reply:
I think it's possible. Just pushed up a modification to the regex which adds the boolean OR in there to handle `//`. 

All these cases are now handled and the resulting toPath is properly outputted after `gatsby build` and `gatsby build --prefix-paths`:

https://facebook.com
http://facebook.com
//facebook.com